server_api/controllers/controller.js

```diff
@@ -0,0 +1,171 @@
+// controller.js
+const UserModel = require('../models/UserModel');
+const ContactModel = require('../models/ContactModel');
+const fs = require('fs');
+const path = require('path');
+const uuid = require('uuid').v1;
+const rootDir = `${path.dirname(require.main.filename)}/public/profile_images/`;
+
+const saveImage = async (ctx) => {
+  const { imageBase64, name } = ctx.request.body;
+  let id = uuid();
+  id = id.substring(0,7);
+  const fileName = `${name}${id}`;
+  const fileExt = 'jpg';
+
+
+  await fs.writeFile(`${rootDir}${fileName}.${fileExt}`, imageBase64, 'base64', function (err) {
+    console.log(err);
+  });
+
+  ctx.body = `${fileName}.${fileExt}`;
+  ctx.status = 200;
+}
+// USERS COLLECTION MODEL
+
+const addUser = async (ctx) => {
+  try {
+    const result = await UserModel.addUser(ctx.request.body);
+    console.log('CREATE USER, ID TO RETUN ON SERVER : ', result._id);
+    ctx.body = result._id;
+    ctx.status = 201;
+  } catch (err) {
+    ctx.status = 500;
+    console.error(err);
+  }
+};
+
+const getAllUsers = async (ctx) => {
+  try {
+    ctx.body = await UserModel.getAllUsers();
+    ctx.status = 200;
+  } catch (err) {
+    ctx.status = 500;
+    console.error(err);
+  }
+};
+
+const updateUser = async (ctx) => {
+  try {
+    await UserModel.updateUser(ctx.request.body);
+    ctx.status = 200;
+  } catch (err) {
+    ctx.status = 500;
+    console.error(err);
+  }
+};
+
```

Comment:
here I believe you are not accounting for many of the secondary conditions, for example, what if the user wanted to delete a contact, bu, the contact that the user wants to delete has already been deleted and the model can't find the user. Your controller will be sending back the wrong response. Have a conditional:

const wasDeleted = UserModel.deleteUser(id);
if(wasDeleted) {
ctx.status = 202;
} else {
ctx.status = 401;
}
and then the the component that is making the request will know how to inform the user for example:
if (res ===401) alert('Contact doesn't exist')
else alert('Contact deleted succesfully')